File: Frontend/src/app/models/assigned-course-user.ts
import {ICourse} from "./course";

export interface IAssignedCourseUser {
  id: string | null,
  professorUserId: string | null,
  courseId: string | null,
}

export interface IAssignedCourseUserResponse {
  id: string | null,
  professorUserId: string | null,
  course: ICourse | null,
}

File: ScheduleFaculty.Core/Entities/AssignedCourseUser.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ScheduleFaculty.Core.Entities;

public class AssignedCourseUser
{
    public Guid Id { get; set; }
    
    [ForeignKey(("ProfessorUser"))]
    public string ProfessorUserId { get; set; }
    public ApplicationUser ProfessorUser { get; set; }
    
    [ForeignKey(("Course"))]
    public Guid CourseId { get; set; }
    public Course Course { get; set; }
}